### PR TITLE
Fix attribute error for _gpus_arg_default loading checkpoint prior to 1.2.8

### LINKED ISSUE #7031: Can't get attribute '_gpus_arg_default'
## 🐛 Bug
when I use the code
```model = model.load_from_checkpoint(ckpt_path)```
It pops out a problem: ```Can't get attribute '_gpus_arg_default' on <module 'pytorch_lightning.utilities.argparse' from '/opt/conda/envs/lasaft/lib/python3.8/site-packages/pytorch_lightning/utilities/argparse.py'>```

### To Reproduce
Very hard to describe the reproduction, because I could not load model.

### Expected behavior
Succeed in loading the model

### Environment

 - PyTorch Version (e.g., 1.0):1.7.1
 - OS (e.g., Linux): Ubuntu 18.04
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source):
 - Python version: 3.8.5
 - CUDA/cuDNN version: 10.2
 - GPU models and configuration: V100
 - Any other relevant information: PL version is 1.3.0rc1

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #7031 

I removed a function in #6898 that I should not have removed. 
Old checkpoints (<1.2.8) have this pickled, so in order to load the checkpoint back we need to have access to the source code. 

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ananthsub — APPROVED** [review]

File: pytorch_lightning/utilities/argparse.py [hunk 1]
```diff
@@ -286,6 +286,13 @@ def _gpus_allowed_type(x) -> Union[int, str]:
         return int(x)
 
 
+def _gpus_arg_default(x) -> Union[int, str]:  # pragma: no-cover
+    # unused, but here for backward compatibility with old checkpoints that need to be able to
+    # unpickle the function from the checkpoint, as it was not filtered out in versions < 1.2.8
+    # see: https://github.com/PyTorchLightning/pytorch-lightning/pull/6898
+    return _gpus_allowed_type(x)
```
Comment (carmocca) [inline pytorch_lightning/utilities/argparse.py:293]: Should we just?:

```suggestion
    pass
```

**Review by carmocca — COMMENTED** [review]
So are we stuck with this function forever? 😢 

Also, why did the legacy checkpoint testing not catch this?

File: pytorch_lightning/utilities/argparse.py (hunk 1 above)
Comment (Borda) [reply pytorch_lightning/utilities/argparse.py:293]: means return None?

**Comment by awaelchli** [comment]
> So are we stuck with this function forever? 😢

I'm afraid so, unless we cut off backward compatibility at some version.
Or maybe there are some pickle tricks we don't know about to avoid it.

Also, why did the legacy checkpoint testing not catch this?
Because all legacy checkpoints were generated without hyperparameters in them.
@Borda we may want to regenerate these checkpoints with hyperparams.

**Review by SeanNaren — APPROVED** [review]

**Comment by Borda** [comment]
> we may want to regenerate these checkpoints with hyperparams.

if you wish to update the model we can regenerate all legacy models...

File: pytorch_lightning/utilities/argparse.py (hunk 1 above)
Comment (carmocca) [reply pytorch_lightning/utilities/argparse.py:293]: yes

Comment (awaelchli) [reply pytorch_lightning/utilities/argparse.py:293]: I'm trying to figure out if that would break something. sigh

In the other PR #6898 , I removed this code
```python
# nothing was passed into the GPUs argument
    if callable(gpus):
        return None
```

So I think when loading the checkpoint, we should directly replace this legacy function with None.

**Review by tchaton — APPROVED** [review]
LGTM !

File: pytorch_lightning/utilities/argparse.py (hunk 1 above)
Comment (carmocca) [reply pytorch_lightning/utilities/argparse.py:293]: @awaelchli tests are green and you have the approvals. Do you want to do this or keep it as is?

Comment (awaelchli) [reply pytorch_lightning/utilities/argparse.py:293]: Ok, I will test once again that I can load the old checkpoint, but with this change.
